src/content/insertable-streams/webgpu/js/multi_video_main.js

```diff
@@ -0,0 +1,274 @@
+        let videoBitmap, gumBitmap;
+        if (videoFrame) {
+            videoBitmap = await createImageBitmap(videoFrame, { resizeWidth: 480, resizeHeight: 270 });
+            device.queue.copyExternalImageToTexture(
+                { source: videoBitmap, origin: { x: 0, y: 0 } },
+                { texture: videoTexture, origin: { x: 0, y: 270 } },
+                {
+                    // the width of the image being copied
+                    width: videoBitmap.width,
+                    height: videoBitmap.height,
+                }
+            );
+            videoBitmap.close();
+            videoFrame.close();
+        }
+        if (gumFrame) {
+            gumBitmap = await createImageBitmap(gumFrame, { resizeWidth: 480, resizeHeight: 270 });
+            device.queue.copyExternalImageToTexture(
+                { source: gumBitmap, origin: { x: 0, y: 0 } },
+                { texture: videoTexture, origin: { x: 480, y: 0 } },
+                {
+                    width: gumBitmap.width,
+                    height: gumBitmap.height,
+                }
+            );
+            gumBitmap.close();
+            gumFrame.close();
+        }
+        const uniformBindGroup = device.createBindGroup({
+            layout: this.renderPipeline_.getBindGroupLayout(0),
+            entries: [
+                {
+                    binding: 0,
+                    resource: this.sampler_,
+                },
+                {
+                    binding: 1,
+                    resource: videoTexture.createView(),
+                },
+            ],
+        });
+
+        const commandEncoder = device.createCommandEncoder();
+        const textureView = this.context_.getCurrentTexture().createView();
+
+        const renderPassDescriptor = {
+            colorAttachments: [
+                {
+                    view: textureView,
+                    loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 1.0 },
+                    storeOp: 'store',
+                },
+            ],
+        };
+        const passEncoder = commandEncoder.beginRenderPass(renderPassDescriptor);
+        passEncoder.setPipeline(this.renderPipeline_);
+        passEncoder.setVertexBuffer(0, this.vertexBuffer_);
+        passEncoder.setBindGroup(0, uniformBindGroup);
+        passEncoder.draw(6, 1, 0, 0);
+        passEncoder.endPass();
+        device.queue.submit([commandEncoder.finish()]);
+        return true;
+    }
+
+
+    async transform(videoStream, gumStream) {
+        const videoSource = videoStream.getReader();
+        const gumSource = gumStream.getReader();
+        if (videoSource === undefined || videoSource === null) {
+            console.log('[WebGPUTransform] videoSource is undefined or null.')
+            return;
+        }
+        if (gumSource === undefined || gumSource === null) {
+            console.log('[WebGPUTransform] gumSource is undefined or null.')
+            return;
+        }
```

Comment:
nit: getReader can't return undefined/null; it would throw if there's a problem.